### PR TITLE
master(dm): clean and treat invalid load task

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
close https://github.com/pingcap/tiflow/issues/3771

### What is changed and how it works?
- if the task in load task etcd KV is stopped, we skip it when hasLoadTaskByWorkerAndSource
- so when start-task, we need to check if a transfer can resolve one load task problem

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix a bug that stop-task during load phase will cause the source always tries to transfer to a worker
```

### CODE REVIEW

File: dm/dm/master/scheduler/scheduler.go [hunk 1]
```diff
@@ -806,6 +806,20 @@ func (s *Scheduler) AddSubTasks(latched bool, cfgs ...config.SubTaskConfig) erro
...
 
+	// 6. check if we can resolve load task by scheduling the source to the worker which has dump files on local disk.
+	for _, cfg := range newCfgs {
+		source := cfg.SourceID
+		worker, ok := s.bounds[source]
+		if !ok {
+			continue
+		}
+
+		if err2 := s.tryResolveLoadTask(worker.BaseInfo().Name, source); err2 != nil {
```
Comment (Ehco1996) [inline dm/dm/master/scheduler/scheduler.go:817]: `AddSubTasks` only hold RLock ,but when `tryResolveLoadTask`  trigger `transfer-source ` we need write lock

File: dm/dm/master/scheduler/scheduler.go [hunk 2]
```diff
@@ -806,6 +806,20 @@ func (s *Scheduler) AddSubTasks(latched bool, cfgs ...config.SubTaskConfig) erro
...
+			s.logger.Error("in AddSubTasks, error when try resolve load task",
```
Comment (Ehco1996) [inline dm/dm/master/scheduler/scheduler.go:818]: why we can mute this error ?

File: dm/dm/master/server.go [hunk 3]
```diff
@@ -1541,7 +1541,10 @@ func (s *Server) removeMetaData(ctx context.Context, taskName, metaSchema string
 	sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS %s",
 		dbutil.TableName(metaSchema, cputil.SyncerOnlineDDL(taskName))))
 
-	_, err = dbConn.ExecuteSQL(ctctx, nil, taskName, sqls)
+	if _, err = dbConn.ExecuteSQL(ctctx, nil, taskName, sqls); err != nil {
+		return err
+	}
+	err = s.scheduler.RemoveLoadTask(taskName)
```
Comment (Ehco1996) [inline dm/dm/master/server.go:1547]: why call `RemoveLoadTask`  again? it already called in  L1509

File: dm/dm/master/scheduler/scheduler.go (hunk 2 above)
Comment (lance6716) [reply dm/dm/master/scheduler/scheduler.go:818]: I think if it's not a big problem if it's failed, the replication can still working for most tasks.

rest comment https://github.com/pingcap/tiflow/pull/4004/commits/9e816129c8f7048a4f265ea999da0b08b72cec24, ptal

File: dm/tests/load_task/run.sh [hunk 4]
```diff
@@ -170,13 +170,94 @@ function test_transfer_two_sources() {
...
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER2_PORT
+
+	# kill worker1, load_task1 will be transferred to worker2, but lack local files
+	ps aux | grep dm-worker1 | awk '{print $2}' | xargs kill || true
+	check_port_offline $WORKER1_PORT 20
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status load_task1" \
+		"different worker in load stage, previous worker: worker1, current worker: worker2" 1
+
+	# now left a load_task KV in etcd
```
Comment (Ehco1996) [inline dm/tests/load_task/run.sh:198]:
```suggestion
	# now stop this task without clean meta (left a load_task KV in etcd)
```

File: dm/dm/master/server.go [hunk 5]
```diff
@@ -502,6 +502,8 @@ func (s *Server) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.S
 			release()
 		}
 
+		go s.scheduler.TryResolveLoadTask(sources)
```
Comment (Ehco1996) [inline dm/dm/master/server.go:505]: already call `TryResolveLoadTask` in  `AddSubTasks` why call again here 🤔

File: dm/tests/load_task/run.sh [hunk 6]
```diff
@@ -170,13 +170,94 @@ function test_transfer_two_sources() {
...
+		"stop-task load_task1" \
+		"\"result\": true" 2
+
+	dmctl_start_task_standalone "$cur/conf/dm-task2-standalone.yaml" "--remove-meta"
+	run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"query-status load_task2" \
+		"\"unit\": \"Sync\"" 1
+
+	# after worker1 goes online, although it has unfinished load_task1, but load_task1 is stopped so should not rebound
+	export GO_FAILPOINTS="github.com/pingcap/tiflow/dm/loader/LoadDataSlowDownByTask=return(\"load_task1\")"
```
Comment (Ehco1996) [inline dm/tests/load_task/run.sh:209]: do we need clean the `GO_FAILPOINTS` env after this test done?

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: dm/tests/load_task/run.sh (hunk 6 above)
Comment (lance6716) [reply dm/tests/load_task/run.sh:209]: I still need this task stuck in load unit so following test can use it

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**